Added more tests for WebProfilerBundle

javiereguiluz opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no <!-- please update src/**/CHANGELOG.md files -->
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? | no <!-- please update UPGRADE-*.md and src/**/CHANGELOG.md files -->
| Tests pass?   | yes    <!-- please add some, will be required by reviewers -->
| Fixed tickets | -   <!-- #-prefixed issue number(s), if any -->
| License       | MIT
| Doc PR        | -

Thanks to @jpauli Code Coverage info about Symfony (http://cov.jpauli.tech/) I found that WebProfiler's controllers are pretty badly covered:

![image](https://user-images.githubusercontent.com/73419/57919817-ec390500-7899-11e9-81b7-763a0b35d0ec.png)

This PR focuses on testing the main controller class:

![image](https://user-images.githubusercontent.com/73419/57919877-04108900-789a-11e9-8a93-3466b672d873.png)

javiereguiluz committed:
Added more tests for WebProfilerBundle

fabpot commented:
@javiereguiluz Can you have a look at the tests, they seem broken.

fabpot commented:
@javiereguiluz Can you have a look at this PR? We cannot merge it as the tests are broken.

nicolas-grekas commented:
Can you please rebase, squash, fix tests?

javiereguiluz commented:
I've rebased and squashed ... but I need help solving a failing test: https://travis-ci.org/symfony/symfony/jobs/560385179#L2602  Thanks.

nicolas-grekas commented:
Can you please rebase so that we can see fresh tests?

javiereguiluz commented:
I've rebased and fixed everything I could, but now I see this error:

> 1) Symfony\Bundle\WebProfilerBundle\Tests\Controller\ProfilerControllerTest::testPanelActionWithWrongPanel
TypeError: Argument 2 passed to Symfony\Bundle\TwigBundle\Controller\ExceptionController::showAction() must be an instance of Symfony\Component\Debug\Exception\FlattenException, instance of Symfony\Component\ErrorRenderer\Exception\FlattenException given, called in /home/travis/build/symfony/symfony/src/Symfony/Bundle/WebProfilerBundle/vendor/symfony/http-kernel/HttpKernel.php on line 146

https://travis-ci.org/symfony/symfony/jobs/590304523#L3318

ro0NL commented on src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php:
```diff
@@ -11,19 +11,110 @@
 {
+    public function testHomeActionWithProfilerDisabled()
+    {
+        $this->expectException(\Symfony\Component\HttpKernel\Exception\NotFoundHttpException::class);
```
already in the use list :angel:

nicolas-grekas commented:
Thank you @javiereguiluz.

nicolas-grekas commented on src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php:
```diff
@@ -11,19 +11,110 @@
+        $crawler = $client->request('GET', '/_profiler/latest?panel=router');
+
+        $this->assertSame('_', $crawler->filter('.metrics .metric .value')->eq(0)->text());
+        $this->assertSame('12', $crawler->filter('.metrics .metric .value')->eq(1)->text());
```
I removed this assertion when merging in 4.4 because it doesn't pass and the patc is painful enough to merge up.
Feel free to submit it again on 4.4 with the proper assertion.
